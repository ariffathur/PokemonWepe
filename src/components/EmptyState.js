import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { FontAwesome } from "@expo/vector-icons";

const EmptyState = (props) => {
  return (
    <View
      style={{
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <FontAwesome name="warning" size={24} color="black" />
      <Text style={[style.desc, { marginTop: 6 }]}>Data tidak tersedia.</Text>
    </View>
  );
};

export default EmptyState;

const style = StyleSheet.create({
  desc: { color: "#45545B", fontSize: 14 },
});
