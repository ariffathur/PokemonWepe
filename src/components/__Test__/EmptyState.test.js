import React from "react";
import renderer from "react-test-renderer";

import EmptyState from "../EmptyState";

describe("<EmptyState />", () => {
  it("has 2 child", () => {
    const tree = renderer.create(<EmptyState />).toJSON();
    expect(tree.children.length).toBe(2);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<EmptyState />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
