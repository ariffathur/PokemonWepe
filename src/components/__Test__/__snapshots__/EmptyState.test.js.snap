// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text />
  <Text
    style={
      Array [
        Object {
          "color": "#45545B",
          "fontSize": 14,
        },
        Object {
          "marginTop": 6,
        },
      ]
    }
  >
    Data tidak tersedia.
  </Text>
</View>
`;
